hashintel/hash
# Add mock-block-dock to all Blocks

## 🌟 What is the purpose of this PR?




Hash has recently published an npm package for mocking an embedding application called [mock-block-dock](https://npmjs.com/package/mock-block-dock), which acts as a testing ground for blocks. This package is also used to wrap the demo blocks in the BlockProtocol website. However, it wasn't used extensively in the local dev setup of the blocks present in the Hash repo.

Thus, this PR adds mock-block-dock as a wrapper and dependency to all existing blocks, allowing us to reuse test functionality across blocks and also catch any errors early.

## 🔍 What does this change?




- Adds `mock-block-dock` as a dev dependency to all blocks.
- Wraps blocks in its wrapper and removes any local block-protocol function implementations.
- Refactors `.js` files to `.tsx` where needed for better code formatting and linting.

## 🛡 What tests cover this?



- Existing linting and TS checks should cover any compiler-side errors.

## ❓ How to test this?



1. Checkout the branch.
2. Run `yarn` on the root directory to install dependencies.
3. Visit each block's directory and run `yarn dev`.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201955153125185
